js/common/lib/session-impl-utils.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {InferenceSession as InferenceSession} from './inference-session.js';
+
+export const processModel = (model: string|ArrayBufferLike|Uint8Array): string|Uint8Array => {
```

Comment:
we can keep the interface simple and just support `string|Uint8Array` instead of `string|ArrayBufferLike|Uint8Array` in the create function

Reply:
Ah, I already have the code ready for createTrainingSession & I wrote it to handle string|ArrayBufferLike|Uint8Array, so simplifying the interface will create more work for me